rqd/rqd/rqnetwork.py

```diff
@@ -206,17 +207,43 @@ def connectGrpcWithRetries(self):
         while True:
             try:
                 self.rqCore.grpcConnected()
+                log.info("Successfully connected to Cuebot.")
+                self.reconnection_attempts = 0
                 break
             except grpc.RpcError as exc:
-                # pylint: disable=no-member
+                # Log the gRPC connection issue
                 if exc.code() == grpc.StatusCode.UNAVAILABLE:
                     log.warning(
-                        'GRPC connection failed. Retrying in %s seconds',
-                        rqd.rqconstants.RQD_GRPC_CONNECTION_ATTEMPT_SLEEP_SEC)
+                        'GRPC connection failed. Retrying in %s seconds (attempt %d)',
+                        rqd.rqconstants.RQD_GRPC_CONNECTION_ATTEMPT_SLEEP_SEC,
+                        self.reconnection_attempts + 1)
+
+                    # Increment reconnection attempts counter
+                    self.reconnection_attempts += 1
+
+                    if self.reconnection_attempts >= rqd.rqconstants.RQD_GRPC_MAX_RETRIES:
+                        log.error("Exceeded maximum reconnection attempts. Marking RQD as wedged.")
+                        self.handle_wedged_state()
+                        break
```

Comment:
First increment the counter than use its value on the warning to avoid two occurrences of `+ 1`

Reply:
Done!